test/syscalls/linux/pty.cc

```diff
@@ -387,6 +388,15 @@ PosixErrorOr<size_t> PollAndReadFd(int fd, void* buf, size_t count,
 }
 
 TEST(PtyTrunc, Truncate) {
+  SKIP_IF(IsRunningWithVFS1() || !IsRunningOnGvisor());
```

Comment:
We shouldn't need the `|| !IsRunningOnGvisor()`. The problem is that the test runner is slightly different between gVisor and native tests, which we can work around by starting the test with this (as is done elsewhere in this file):

```cpp
SKIP_IF(IsRunningWithVFS1());

// setsid either puts us in a new session or fails because we're already the
// session leader. Either way, this ensures we're the session leader and have
// no controlling terminal.
setsid();
```

It should now pass on native and VFS2:
```bash
bazel test //test/syscalls:pty_test_{native,runsc_ptrace_vfs2} --test_filter=PtyTrunc.Truncate
```